src/main/java/org/jabref/logic/importer/fetcher/IacrEprintFetcher.java

```diff
@@ -0,0 +1,116 @@
+        if (!IDENTIFIER_REGEX.test(identifier)) {
+            throw new FetcherException("Wrong identifier format");
+        }
+        String downloaded;
+        try {
+            URL url = new URL("https://eprint.iacr.org/eprint-bin/cite.pl?entry=" + identifier);
+            URLDownload download = new URLDownload(url);
+            downloaded = download.asString();
+        } catch (IOException e) {
+            FetcherException ex = new FetcherException(e.getMessage());
+            ex.addSuppressed(e);
+            throw ex;
+        }
+
+        String entryString = downloaded.substring(downloaded.indexOf("<PRE>") + 5, downloaded.indexOf("</PRE>")).trim();
+        try {
+            Optional<BibEntry> entry = BibtexParser.singleFromString(entryString, prefs);
+            if (entry.isPresent()) {
+                setAdditionalFields(entry.get(), identifier);
+            }
+            return entry;
+        } catch (IOException | ParseException e) {
+            throw new FetcherException(e.toString());
+        }
+    }
+
+    @Override
+    public String getName() {
+        return "IACR eprint";
+    }
+
+    private void setAdditionalFields(BibEntry entry, String identifier) throws IOException {
+        String url = "https://eprint.iacr.org/" + identifier;
+        entry.setField(FieldName.URL, url);
+
+        URLDownload download = new URLDownload(url);
+        String content = download.asString();
+
+        String abstractText = getValueBetween("<b>Abstract: </b>", "<p />", content);
+        // for some reason, all spaces are doubled...
+        abstractText = abstractText.replaceAll("\\s(\\s)", "$1");
+        entry.setField(FieldName.ABSTRACT, abstractText);
+
+        String startOfVersionString = "<b>Version: </b><a href=\"/" + identifier + "/";
+        String version = getValueBetween(startOfVersionString, "\"", content);
+        entry.setField(FieldName.VERSION, version);
+
+        String dates = getValueBetween("<b>Date: </b>", "<p />", content);
+        entry.setField(FieldName.DATE, getDate(dates));
+    }
+
+    public static String getDate(String dateContent) {
+        String[] rawDates = dateContent.split(",");
+        List<String> formattedDates = new ArrayList<>();
+
+
+        for (String rawDate : rawDates) {
+            try {
+                rawDate = rawDate.trim();
+                Matcher dateMatcher = Pattern.compile("[a-z ]+(\\d{1,2} [A-Za-z][a-z]{2} \\d{4})").matcher(rawDate);
+                if (dateMatcher.find()) {
+                    Date date = DATE_FORMAT_WEBSITE.parse(dateMatcher.group(1));
+                    formattedDates.add(DATE_FORMAT_BIBTEX.format(date));
+                }
+            } catch (Exception e) {
+                // Just skip
+            }
+        }
+
+        Collections.sort(formattedDates, Collections.reverseOrder());
+        return formattedDates.get(0);
+    }
+
+    private static String getValueBetween(String from, String to, String haystack) {
```

Comment:
You could see if there is already a method in StringUtils

Reply:
Thanks, there was indeed one. Keeping the functionality in a separate method though to have the error handling (nothing found) in one place.